llvm/lib/Target/RISCV/RISCVSchedXiangShanKunMingHu.td

```diff
@@ -0,0 +1,1489 @@
+  }
+}
+
+// VFCVT
+foreach mx = SchedMxListFW in {
+  foreach sew = NoZvfhSchedSEWSet_rm8and16<mx, isF=1, isWidening=1>.val in {
+    defvar Cycles = XSGetCyclesVFCVT<mx>.c;
+    defvar IsWorstCase = XSIsWorstCaseMXSEW<mx, sew, SchedMxListFW, isF=1>.c;
+    let Latency = Cycles in {
+      defm "" : LMULSEWWriteResMXSEW<"WriteVFWCvtFToFV",   [XSPipVFEX1], mx, sew, IsWorstCase>;
+      defm "" : LMULSEWWriteResMXSEW<"WriteVFNCvtIToFV",   [XSPipVFEX1], mx, sew, IsWorstCase>;
+      defm "" : LMULSEWWriteResMXSEW<"WriteVFNCvtFToFV",   [XSPipVFEX1], mx, sew, IsWorstCase>;
+    }
+  }
+}
+
+// VFCVT
+foreach mx = SchedMxListW in {
+  defvar Cycles = XSGetCyclesVFCVT<mx>.c;
+  defvar IsWorstCase = XSIsWorstCaseMX<mx, SchedMxListW>.c;
+  let Latency = Cycles in {
+    defm "" : LMULWriteResMX<"WriteVFNCvtFToIV",   [XSPipVFEX1], mx, IsWorstCase>;
+  }
+}
+
+// 14. Vector Reduction Operations
+// VIPU
+foreach mx = SchedMxList in {
+  foreach sew = SchedSEWSet<mx>.val in {
+    defvar Cycles = XSGetCyclesVIPU<mx>.c;
+    defvar IsWorstCase = XSIsWorstCaseMXSEW<mx, sew, SchedMxList>.c;
+    let Latency = Cycles in {
+      defm "" : LMULSEWWriteResMXSEW<"WriteVIRedV_From",        [XSPipVFEX1], mx, sew, IsWorstCase>;
+      defm "" : LMULSEWWriteResMXSEW<"WriteVIRedMinMaxV_From",  [XSPipVFEX1], mx, sew, IsWorstCase>;
+    }
+  }
+}
+
+// VIPU
+foreach mx = SchedMxListWRed in {
+  foreach sew = SchedSEWSet<mx, isF=0, isWidening=1>.val in {
+    defvar Cycles = XSGetCyclesVIPU<mx>.c;
+    defvar IsWorstCase = XSIsWorstCaseMXSEW<mx, sew, SchedMxListWRed>.c;
+    let Latency = Cycles in {
+      defm "" : LMULSEWWriteResMXSEW<"WriteVIWRedV_From",       [XSPipVFEX1], mx, sew, IsWorstCase>;
+    }
+  }
+}
+
+// VFALU
+foreach mx = SchedMxListF in {
+  foreach sew = NoZvfhSchedSEWSet_rm8and16<mx, isF=1, isWidening=0>.val in {
+    defvar Cycles = XSGetCyclesVFALU<mx>.c;
+    defvar IsWorstCase = XSIsWorstCaseMXSEW<mx, sew, SchedMxListF, isF=1>.c;
+    let Latency = Cycles in {
+      defm "" : LMULSEWWriteResMXSEW<"WriteVFRedV_From",        [XSPipeGroupVFALU], mx, sew, IsWorstCase>;
+      defm "" : LMULSEWWriteResMXSEW<"WriteVFRedOV_From",       [XSPipeGroupVFALU], mx, sew, IsWorstCase>;
+      defm "" : LMULSEWWriteResMXSEW<"WriteVFRedMinMaxV_From",  [XSPipeGroupVFALU], mx, sew, IsWorstCase>;
+    }
+  }
+}
+
+// VFALU
+foreach mx = SchedMxListFWRed in {
+  foreach sew = NoZvfhSchedSEWSet_rm8and16<mx, isF=1, isWidening=1>.val in {
+    defvar Cycles = XSGetCyclesVFALU<mx>.c;
+    defvar IsWorstCase = XSIsWorstCaseMXSEW<mx, sew, SchedMxListFWRed, isF=1>.c;
+    let Latency = Cycles in {
+      defm "" : LMULSEWWriteResMXSEW<"WriteVFWRedV_From",       [XSPipeGroupVFALU], mx, sew, IsWorstCase>;
+      defm "" : LMULSEWWriteResMXSEW<"WriteVFWRedOV_From",      [XSPipeGroupVFALU], mx, sew, IsWorstCase>;
+    }
+  }
+}
+
+// 15. Vector Mask Instructions
+// VIALU
+foreach mx = SchedMxList in {
+  defvar Cycles = XSGetCyclesVIALU<mx>.c;
```

Comment:
Are you sure this is correct? Since masks always fit into a single LMUL=1 vector register, you'd expect that an LMUL=8 SEW=8 vmand.mm would have the same latency has a LMUL=1 vand.vv. Or does xiangshan use a different internal format for mask registers? See how the SiFivdP600 schedular sets the latency of all mask instructions to 1.